torch/distributed/_functional_collectives.py

```diff
@@ -616,3 +621,24 @@ def reduce_scatter_tensor_inplace(
     legacy_allgather: all_gather_tensor_inplace,
     legacy_reducescatter: reduce_scatter_tensor_inplace,
 }
+
+def _all_gather_into_tensor_coalesced_fallback(output_tensors, input_tensors, group, async_op=False):
+    # all_gather_coalesced is useless, it doesn't work under NCCL and does lots of copies under Gloo
+    # all_gather is useless too because it's single tensor
+    # NCCL's PG::all_gather with multiple tensors is broken, it only works for the multi-device setting
+    #  and fails if you mix same-size with different-size tensor lists.
+    # _coalescing_manager crashed NCCL when used with all_gather_into_tensor.
+    work_list = []
+    if input_tensors[0].is_cpu:
```

Comment:
is this fallback here for "cpu" only as there's no "tensor" based allgather? (i.e. allgather_into_tensor)?